src/utils.jl

```diff
@@ -0,0 +1,22 @@
+"""
+    tcount(c)
+
+Returns the number of non-Clifford gates in circuit `c`. 
+"""
+tcount(bir::BlockIR) = tcount(bir.circuit)
+tcount(c::Chain) = sum(tcount(g) for g in c.args)
+tcount(::Routine) = error("T-count for type $T is not defined")
+tcount(g::AdjointOperation) = tcount(g.parent)
+function tcount(g::Ctrl)
+    if length(g.ctrl) == 1 && length(g.gate.locations) == 1
+        (g.gate.operation === X || g.gate.operation === Z) && return 0
+    end
+    return tcount(decompose_zx(g))
+end
+tcount(g::Gate) = tcount(g.operation) * length(g.locations)
+tcount(::IntrinsicRoutine) = 0
+tcount(::TGate) = 1
+tcount(g::Rx) = (abs(rem2pi(g.θ*4, RoundNearest)) < 1e-10 ? 0 : 1)
```

Comment:
I think the 1e-10 bound Jinguo uses was incorrect, the floating-point precision should be `sqrt(eps(Float64))` which is around `1e-8`

Reply:
we need to check this based on the type of theta